provider/clusterautoscaler/cluster_autoscaler_resource.go

```diff
@@ -0,0 +1,640 @@
+			fmt.Sprintf(
+				"Failed creating autoscaler for cluster '%s': %v",
+				plan.Cluster.ValueString(), err,
+			),
+		)
+		return
+	}
+
+	diags = response.State.Set(ctx, plan)
+	response.Diagnostics.Append(diags...)
+}
+
+func (r *ClusterAutoscalerResource) Read(ctx context.Context, request resource.ReadRequest, response *resource.ReadResponse) {
+	state := &ClusterAutoscalerState{}
+	diags := request.State.Get(ctx, state)
+	response.Diagnostics.Append(diags...)
+	if response.Diagnostics.HasError() {
+		return
+	}
+
+	getResponse, err := r.collection.Cluster(state.Cluster.ValueString()).Autoscaler().Get().SendContext(ctx)
+	if err != nil && getResponse.Status() == http.StatusNotFound {
+		tflog.Warn(ctx, fmt.Sprintf("autoscaler for cluster (%s) not found, removing from state",
+			state.Cluster.ValueString(),
+		))
+		response.State.RemoveResource(ctx)
+		return
+
+	} else if err != nil {
+		response.Diagnostics.AddError(
+			"Failed getting cluster autoscaler",
+			fmt.Sprintf(
+				"Failed getting autoscaler for cluster '%s': %v",
+				state.Cluster.ValueString(), err,
+			),
+		)
+		return
+	}
+
+	response.Diagnostics.Append(diags...)
+	if response.Diagnostics.HasError() {
+		return
+	}
+
+	populateAutoscalerState(getResponse.Body(), state.Cluster.ValueString(), state)
+	diags = response.State.Set(ctx, state)
+	response.Diagnostics.Append(diags...)
+}
+
+func (r *ClusterAutoscalerResource) Update(ctx context.Context, request resource.UpdateRequest,
+	response *resource.UpdateResponse) {
+	var diags diag.Diagnostics
+
+	// Get the plan:
+	plan := &ClusterAutoscalerState{}
+	diags = request.Plan.Get(ctx, plan)
+	response.Diagnostics.Append(diags...)
+	if response.Diagnostics.HasError() {
+		return
+	}
+
+	_, err := r.collection.Cluster(plan.Cluster.ValueString()).Autoscaler().Get().SendContext(ctx)
+
+	if err != nil {
+		response.Diagnostics.AddError(
+			"Failed getting cluster autoscaler",
+			fmt.Sprintf(
+				"Failed getting autoscaler for cluster '%s': %v",
+				plan.Cluster.ValueString(), err,
+			),
+		)
+		return
+	}
+
+	autoscaler, err := clusterAutoscalerStateToObject(plan)
+	if err != nil {
+		response.Diagnostics.AddError(
+			"Failed updating cluster autoscaler",
+			fmt.Sprintf(
+				"Failed updating autoscaler for cluster '%s: %v ",
+				plan.Cluster.ValueString(), err,
+			),
+		)
+		return
+	}
+
+	update, err := r.collection.Cluster(plan.Cluster.ValueString()).
+		Autoscaler().Update().Body(autoscaler).SendContext(ctx)
+	if err != nil {
+		response.Diagnostics.AddError(
+			"Failed updating cluster autoscaler",
+			fmt.Sprintf(
+				"Failed updating autoscaler for cluster '%s': %v",
+				plan.Cluster.ValueString(), err,
+			),
+		)
+		return
+	}
+
+	object := update.Body()
+	state := &ClusterAutoscalerState{}
+	populateAutoscalerState(object, plan.Cluster.ValueString(), state)
+
+	diags = response.State.Set(ctx, state)
+	response.Diagnostics.Append(diags...)
+	if response.Diagnostics.HasError() {
```

Comment:
why do we need that if statement?

Reply:
if we already have errors we should exit.